src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -10,115 +10,100 @@ namespace System
+        /// Initializes new instance of <see cref="TimeProvider"/>.
         /// </summary>
-        /// <exception cref="ArgumentOutOfRangeException">The value of <paramref name="timestampFrequency"/> is negative or zero.</exception>
-        /// <param name="timestampFrequency">Frequency of the values returned from <see cref="GetTimestamp"/> method.</param>
-        protected TimeProvider(long timestampFrequency)
+        protected TimeProvider()
         {
-#if SYSTEM_PRIVATE_CORELIB
-            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(timestampFrequency);
-#else
-            if (timestampFrequency <= 0)
-            {
-                throw new ArgumentOutOfRangeException(nameof(timestampFrequency), timestampFrequency, SR.Format(SR.ArgumentOutOfRange_Generic_MustBeNonNegativeNonZero, nameof(timestampFrequency)));
-            }
-#endif // SYSTEM_PRIVATE_CORELIB
-
-            TimestampFrequency = timestampFrequency;
-            _timeToTicksRatio = (double)TimeSpan.TicksPerSecond / TimestampFrequency;
         }
 
         /// <summary>
         /// Gets a <see cref="DateTimeOffset"/> value whose date and time are set to the current
         /// Coordinated Universal Time (UTC) date and time and whose offset is Zero,
         /// all according to this <see cref="TimeProvider"/>'s notion of time.
         /// </summary>
-        public abstract DateTimeOffset UtcNow { get; }
+        /// <remarks>
+        /// The default implementation returns <see cref="DateTimeOffset.UtcNow"/>.
+        /// </remarks>
+        public virtual DateTimeOffset GetUtcNow() => DateTimeOffset.UtcNow;
 
         private static readonly long s_minDateTicks = DateTime.MinValue.Ticks;
         private static readonly long s_maxDateTicks = DateTime.MaxValue.Ticks;
 
         /// <summary>
         /// Gets a <see cref="DateTimeOffset"/> value that is set to the current date and time according to this <see cref="TimeProvider"/>'s
-        /// notion of time based on <see cref="UtcNow"/>, with the offset set to the <see cref="LocalTimeZone"/>'s offset from Coordinated Universal Time (UTC).
+        /// notion of time based on <see cref="GetUtcNow"/>, with the offset set to the <see cref="LocalTimeZone"/>'s offset from Coordinated Universal Time (UTC).
         /// </summary>
-        public DateTimeOffset LocalNow
+        public DateTimeOffset GetLocalNow()
         {
-            get
-            {
-                DateTime utcDateTime = UtcNow.UtcDateTime;
-                TimeSpan offset = LocalTimeZone.GetUtcOffset(utcDateTime);
-
-                long localTicks = utcDateTime.Ticks + offset.Ticks;
-                if ((ulong)localTicks > (ulong)s_maxDateTicks)
-                {
-                    localTicks = localTicks < s_minDateTicks ? s_minDateTicks : s_maxDateTicks;
-                }
+            DateTimeOffset utcDateTime = GetUtcNow();
+            TimeSpan offset = (LocalTimeZone ?? TimeZoneInfo.Local).GetUtcOffset(utcDateTime);
```

Comment:
This trying to protect against a derived type erroneously returning null from LocalTimeZone?  If we want to protect against that, I think we should throw an exception instead... otherwise it suggests returning null is acceptable.